contracts/Checkpointing.sol

```diff
@@ -62,8 +66,10 @@ library Checkpointing {
     }
 
     function lastUpdated(History storage self) internal view returns (uint256) {
-        if (self.history.length > 0) {
-            return uint256(self.history[self.history.length - 1].time);
+        uint256 length = self.history.length;
+
+        if (length > 0) {
+            return uint256(self.history[length - 1].time);
```

Comment:
I this case I think we are copying a storage value to memory and then from memory to the stack, I'm not sure if this will make the compiler optimizing it in a way to use less gas than before. I would expect the compiler to load the length value to the stack once and dup it as much times as it needs it.

Reply:
Same here, I tried with Remix and the results were slightly better:
https://gist.github.com/bingen/1d5197a984b6dd69b1b1f3d5dd0e84b0
As in this case I don't think code gets worse, if you agree I'll leave it.